src/graph.py

```diff
@@ -345,16 +353,19 @@ def output_forest_as_networkx(self, vertex_indices, edge_indices):
             networkx.Graph: a networkx graph object
         """
 
+        if len(vertex_indices) == 0: 
+            return nx.empty_graph(0), nx.empty_graph(0)
+
         # Replace the edge indices with the actual edges (source name, target name) by indexing into the interactome
         edges = self.interactome_dataframe.loc[edge_indices]
         forest = nx.from_pandas_edgelist(edges, 'source', 'target', edge_attr=True)
-        # the above won't capture the singletons, so we'll add them here
+        # the above won't capture the singletons, so we'll add them here 
         forest.add_nodes_from(list(set(self.nodes[vertex_indices]) - set(forest.nodes())))
 
         # Set node degrees as attributes on nodes in the netowrkx graph
         nx.set_node_attributes(forest, pd.DataFrame(self.node_degrees, index=self.nodes, columns=['degree']).loc[list(forest.nodes())].astype(int).to_dict(orient='index'))
 
-        # Set all othe attributes on graph
+        # Set all the attributes on graph
```

Comment:
This is a little confusing. Line 366 sets node degrees on the graph, and then you set it again with the following line, since degrees are now a column in `node_attributes`.

Reply:
I didn't write this. It exists in master currently.

Reply:
391b84d1289d38bcc6d3ba7a826a8924520ddbf6